server/chat-plugins/chatlog.ts

```diff
@@ -545,12 +545,12 @@ export const LogSearcher = new class {
 			});
 			results = stdout.split('--');
 		} catch (e) {
-			if (e.stdout || e.message.includes('stdout maxBuffer')) {
+			if (e.code === 2) throw e; // 2 means an error in ripgrep
```

Comment:
Is 2 the "no results found" error code? Shouldn't it be the other way around?

Reply:
2 is "there is an error with ripgrep that _isn't_ no results", which should be thrown.
Otherwise, e.stdout might exist, which is handled, or there will be no results - also handled.

Reply:
What's the error code if there _is_ no result? That's the one that should be special-cased.

Reply:
no result is `1`, but it seems like it would be easier to just handle the actual error and then check stdout, since that also covers things like the weird issues with execFile that we had last time?

Reply:
`if (e.code !== 1)` seems safer.

As usual: It's much better to crash if you don't know what's going on, than to not crash and have a bug that's very difficult to track down.